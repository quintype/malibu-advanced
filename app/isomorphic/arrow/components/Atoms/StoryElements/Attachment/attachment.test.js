import * as React from "react";
import { shallow } from "enzyme";
import { generateStoryElementData } from "../../../Fixture";
import { Attachment } from "./index";

let element = generateStoryElementData("attachment");

describe("Attachment Story Element", () => {
  it("Should render default template", () => {
    const wrapper = shallow(<Attachment element={element} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should not render default template if url is not passed", () => {
    const wrapper = shallow(<Attachment element={{ ...element, url: "" }} />);
    expect(wrapper.find({ "data-test-id": "attachment" }).length).toEqual(0);
  });

  it("Should not render default template if content-type is not passed", () => {
    const wrapper = shallow(<Attachment element={{ ...element, "content-type": "" }} />);
    expect(wrapper.find({ "data-test-id": "attachment" }).length).toEqual(0);
  });

  it("Should render default template with default file-name if not passed", () => {
    const wrapper = shallow(<Attachment element={{ ...element, "file-name": "" }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render PDF icon if content-type is pdf", () => {
    const wrapper = shallow(<Attachment element={element} />);
    expect(wrapper.find({ "data-test-id": "doc" }).length).toEqual(0);
  });

  it("Should render DOC icon if content-type is doc", () => {
    element = { ...element, "content-type": "application/doc" };
    const wrapper = shallow(<Attachment element={element} />);
    expect(wrapper.find({ "data-test-id": "pdf" }).length).toEqual(0);
  });
});
