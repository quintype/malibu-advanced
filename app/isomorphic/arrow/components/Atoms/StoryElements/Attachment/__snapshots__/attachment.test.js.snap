// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment Story Element Should render default template 1`] = `
<AttachmentBase
  element={
    Object {
      "content-type": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
      "description": "",
      "family-id": "95bf1052-d663-446d-ace7-d014325e0027",
      "file-name": "document__7_ (1).docx",
      "id": "2f648c29-ef8f-442c-903a-263f82e631dd",
      "metadata": Object {
        "file-size": 8209,
      },
      "page-url": "/story/7155b5c2-80a4-4922-bdf1-73f1ff04311e/element/2f648c29-ef8f-442c-903a-263f82e631dd",
      "s3-key": "ace/2019-08/21f3a19b-1d98-46bc-95e3-a5bdc9045ae9/document__7___1_.docx",
      "subtype": "attachment",
      "title": "",
      "type": "file",
      "url": "https://thumbor-stg.assettype.com/ace/2019-08/21f3a19b-1d98-46bc-95e3-a5bdc9045ae9/document__7___1_.docx",
    }
  }
/>
`;

exports[`Attachment Story Element Should render default template with default file-name if not passed 1`] = `
<AttachmentBase
  element={
    Object {
      "content-type": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
      "description": "",
      "family-id": "95bf1052-d663-446d-ace7-d014325e0027",
      "file-name": "",
      "id": "2f648c29-ef8f-442c-903a-263f82e631dd",
      "metadata": Object {
        "file-size": 8209,
      },
      "page-url": "/story/7155b5c2-80a4-4922-bdf1-73f1ff04311e/element/2f648c29-ef8f-442c-903a-263f82e631dd",
      "s3-key": "ace/2019-08/21f3a19b-1d98-46bc-95e3-a5bdc9045ae9/document__7___1_.docx",
      "subtype": "attachment",
      "title": "",
      "type": "file",
      "url": "https://thumbor-stg.assettype.com/ace/2019-08/21f3a19b-1d98-46bc-95e3-a5bdc9045ae9/document__7___1_.docx",
    }
  }
/>
`;
